### PR TITLE
19 create time attendance input

### PR DESCRIPTION
**Pull request by SvetlanaNagovicyna**

### CODE REVIEW

**Comment by SvetlanaNagovicyna** [comment]
Screenshot_2

**Comment by SvetlanaNagovicyna** [comment]
Screenshot_3

**Comment by SvetlanaNagovicyna** [comment]
image

File: src/app/private/components/time/time.component.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    return this.times;
+  }
+
+  addTime(type: Time): void {
+    const timeData: Times = this.generateTimesData(type);
+    this.timeService.addTime(timeData, this.userService.user?.id).subscribe();
+  }
+
+  addCameTime(): void {
+    if (this.currentTime.cameTime) return;
```
Comment (leonxaoss) [inline src/app/private/components/time/time.component.ts:76]: please use {} for if

File: src/app/private/components/time/time.component.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  continueWork(): void {
+    if (!this.currentTime.leaveTime) return;
+    this.currentTime.leaveTime = '';
+    this.addTime('leaveTime');
+  }
+
+  getCurrentTime(): string {
+    return new Date().toLocaleTimeString([], {hour: '2-digit', minute: '2-digit'});
```
Comment (leonxaoss) [inline src/app/private/components/time/time.component.ts:99]: let's save whole Date in the database, not only time

File: src/app/private/components/time/time.component.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    this.isShowInput = !this.isShowInput;
+  }
+
+  generateTimesData(type: Time): Times {
+    this.times[this.currentDate] = {
+      ...this.times[this.currentDate],
+      [type]: this.currentTime[type]
+    };
```
Comment (leonxaoss) [inline src/app/private/components/time/time.component.ts:65]: need to rework this part a bit, there is interface difference

File: src/app/private/components/time/time.component.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  addTime(type: Time): void {
+    const timeData: Times = this.generateTimesData(type);
+    this.timeService.addTime(timeData, this.userService.user?.id).subscribe();
+  }
+
+  addCameTime(): void {
+    if (this.currentTime.cameTime) return;
+    this.currentTime.cameTime = this.getCurrentTime();
+    this.addTime('cameTime');
```
Comment (leonxaoss) [inline src/app/private/components/time/time.component.ts:78]: let's create enum for types

**Comment by SvetlanaNagovicyna** [comment]
image

File: src/app/private/components/time/time.component.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    };
+
+    return this.currentTime;
+  }
+
+  addTime(type: TimeState): void {
+    const timeData: Times = this.generateTimesData(type);
+    this.timeService.addTime(timeData, this.userService.user?.id, this.currentDate).subscribe({
+      next: (res: Times): void => {
+        if (res.lunchTime) {
```
Comment (leonxaoss) [inline src/app/private/components/time/time.component.ts:73]: you should check type if it's lunchTime)

**Pull request merged**